.github/workflows/cuda-test.yml

```diff
@@ -0,0 +1,15 @@
+name: Docker Image CI
+on:
+  workflow_dispatch: null
+  push:
+    paths:
+      - tests/ml-test.Dockerfile
+      - .github/workflows/cuda-test.yml
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions: write-all
```

Comment:
`write-all` is needed here?

Reply:
good question, let's try without that.

Reply:
Oh, I'm sorry. What I was trying to say was that read permission would be sufficient.

Reply:
But I don't think this test is ever using the GITHUB_TOKEN here anyway?  I think I had left write-all in by mistake from a previous action that was also pushing to github (and even then it should plausibly have a more scoped write permission).  I think the test is happy without this.

Reply:
Since this repository is old, I assume write-all privileges are granted by default, but if you follow best practices, you should grant only minimal privileges.
Since we are only reading files here, we expect read permission to be sufficient.

Reply:
oh good to know! I didn't realize they did legacy privileges but of course that makes sense or actions would break.  I've gone with read-all now.